### ISSUE TITLE
Error while running the script

### ISSUE THREAD

**Issue by hyukchan** [jamestut/awdlkiller#1]
Hi,

I'm having high ping issues on my Mac,
and I found your application from the net.

But while running the awdlkiller app, i'm having this error :

awdlkiller: Error initial disable interface: Operation not permitted

even tho I'm running it in sudo.

Do you have any idea why ?

Thanks for your help

**Comment by jamestut** [comment]
Hi there, can you paste here the full command and the output that you get?

**Comment by hyukchan** [comment]
Hi ! 
So when i run with launchctl, i get this:

➜ sudo launchctl load -w /Library/LaunchDaemons/jamestut.awdlkiller.plist

Password:
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.

Then i tried to run the script directly:

➜ /usr/local/bin/awdlkiller
awdlkiller: Error escalating permission to root. Either run this app as root or set setuid bit with root permission.

and with "sudo": 

➜ sudo /usr/local/bin/awdlkiller
awdlkiller: Error initial disable interface: Operation not permitted

Thanks for your help

**Comment by fatdoyy** [comment]
I'm having the same error.  Did you solved the issue already? @hyukchan

**Comment by jamestut** [comment]
Hi @hyukchan and @fatdoyy, I just had a brand new macOS installation (fresh install and nothing else installed, SIP on) from 2 different M1 MacBooks and I didn't run into this issue. I couldn't manage to reproduce the issue that you described here.

Wondering what will happen if you do `sudo ifconfig awdl0 down` manually? And then use that same terminal to `sudo ./awdlkiller`?

**Comment by fatdoyy** [comment]
Hi @jamestut, running `sudo ifconfig awdl0 down` is fine but running `sudo ./awdlkiller` gave the same error as above:  `awdlkiller: Error initial disable interface: Operation not permitted`. 

I'm thinking maybe there's some installation error on my side. May I ask where exactly should I put the `manager.py` file?

**Comment by jamestut** [comment]
Hi @fatdoyy, the `manager.py` itself is not a requirement to run the `awdlkiller`. You can simply put the `awdlkiller` anywhere and it should work straight away without that permission error.

Once you've done installing the `jamestut.awdlkiller.plist` file to `/Library/LaunchDaemon`, you can put the `manager.py` in `/usr/local/bin` and rename it to any file that you want, for example `awdl` for easy management of the `awdlkiller`.

Also, if I got the time, I'll make an install script around 2 weeks from now or so.

**Comment by dholt** [comment]
make sure you do `sudo chmod u+s awdlkiller`

**Comment by jamestut** [comment]
I've tested this on a freshly installed macOS on Apple Silicon with SIP and without anything installed and not iCloud logged-in, and running `sudo ./awdlkiller` works straight away. Additionally, now I've provided a rudimentary installer in this repo (and that installer also works on the aforementioned clean installation of macOS, but with CLDT installed).

Closing this issue as it is highly likely some 3rd party security software might interfere with the operation of this program.
